### ISSUE TITLE
Flaky test TestKeyManagerContract/GetKey/rsa-4096

### ISSUE THREAD

**Issue by rturner3** [spiffe/spire#3339]
```
--- FAIL: TestKeyManagerContract (47.62s)
    --- FAIL: TestKeyManagerContract/GetKey (32.75s)
        --- FAIL: TestKeyManagerContract/GetKey/rsa-4096 (30.04s)
            keymanagertest.go:219: 
                	Error Trace:	/Users/runner/work/spire/spire/pkg/server/plugin/keymanager/disk/keymanagertest.go:219
                	            				/Users/runner/work/spire/spire/pkg/server/plugin/keymanager/disk/keymanagertest.go:177
                	Error:      	Received unexpected error:
                	            	rpc error: code = DeadlineExceeded desc = keymanager(disk): context deadline exceeded
                	Test:       	TestKeyManagerContract/GetKey/rsa-4096
FAIL
FAIL	github.com/spiffe/spire/pkg/server/plugin/keymanager/disk	47.868s
```

Build link: https://github.com/spiffe/spire/runs/7775372796?check_suite_focus=true

**Comment by azdagron** [comment]
Yeah this is an unfortunate consequence of that test actually generating a 4096 bit RSA key. This can take a while on resource constrained/shared environments like CI/CD. Enough that it is triggering a timeout. Not _quite_ sure what we should do here.

**Comment by rturner3** [comment]
This test failure also looks related:
```
--- FAIL: TestKeyManagerContract (47.01s)
    --- FAIL: TestKeyManagerContract/GetKeys (31.98s)
        keymanagertest.go:219: 
            	Error Trace:	/Users/runner/work/spire/spire/pkg/server/plugin/keymanager/disk/keymanagertest.go:219
            	            				/Users/runner/work/spire/spire/pkg/server/plugin/keymanager/disk/keymanagertest.go:202
            	            				/Users/runner/work/spire/spire/pkg/server/plugin/keymanager/disk/keymanagertest.go:134
            	Error:      	Received unexpected error:
            	            	rpc error: code = DeadlineExceeded desc = keymanager(disk): context deadline exceeded
            	Test:       	TestKeyManagerContract/GetKeys
FAIL
FAIL	github.com/spiffe/spire/pkg/server/plugin/keymanager/disk	47.262s
```

**Comment by azdagron** [comment]
This should be fixed by #3499.
